File: src/app/model/star.ts
import { Line } from "./line"
import { Point } from "./point"
import { Equatable } from "./equatable"
import { Copyable } from "./copyable";

export class Star extends Point implements Equatable<Star>, Copyable<Star> {
    line: Line

    constructor(x: number, y: number, length: number, width: number, color: string = '#74B6F6') {
        super(x, y, 10, color);
        this.line = new Line(length, width);
    }

    public equals(star: Star): boolean {
        if (!super.equals(star)) { return false; }
        if (!this.line.equals(star.line)) { return false; }
        return true;
    }

    public copy(): Star {
        return new Star(this.x, this.y, this.line.length, this.line.width, this.color)
    }
}

File: astrograph/model/star.py
from model.point import Point
from model.line import Line


class Star(Point):
    x: int
    y: int
    line: Line

    def __init__(self, x: int, y: int, length: int, width: int):
        Point.__init__(self, x, y, 5)

        self.x = x
        self.y = y
        self.line = Line(length, width)

    def dictionary(self):
        return {"x": self.x, "y": self.y, "line": self.line.dictionary()}
